toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagegen/diskutils"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/file"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+)
+
+func enableVerityPartition(imageChroot *safechroot.Chroot) error {
+	var err error
+
+	// Integrate systemd veritysetup dracut module into initramfs img.
+	systemdVerityDracutModule := "systemd-veritysetup"
+	err = buildDracutModule(systemdVerityDracutModule, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	// Update mariner config file with the new generated initramfs file.
+	err = updateMarinerCfgWithInitramfs(imageChroot)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildDracutModule(dracutModuleName string, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	listKernels := func() ([]string, error) {
+		var kernels []string
+		// Use RootDir to get the path on the host OS
+		bootDir := filepath.Join(imageChroot.RootDir(), "boot")
+		files, err := filepath.Glob(filepath.Join(bootDir, "vmlinuz-*"))
+		if err != nil {
+			return nil, err
+		}
+		for _, file := range files {
+			kernels = append(kernels, filepath.Base(file))
+		}
+		return kernels, nil
+	}
+
+	kernelFiles, err := listKernels()
+	if err != nil {
+		return fmt.Errorf("failed to list kernels: %w", err)
+	}
+
+	if len(kernelFiles) == 0 {
+		return fmt.Errorf("no kernels found in chroot environment")
+	}
+
+	// Check if more than one kernel is found
+	if len(kernelFiles) > 1 {
+		return fmt.Errorf("multiple kernels found in chroot environment, expected only one")
+	}
+
+	// Extract the version from the kernel filename (e.g., vmlinuz-5.15.131.1-2.cm2 -> 5.15.131.1-2.cm2)
+	kernelVersion := strings.TrimPrefix(kernelFiles[0], "vmlinuz-")
+
+	err = imageChroot.Run(func() error {
+		// TODO: Config Dracut module systemd-veritysetup - task 6421.
+		err = shell.ExecuteLiveWithErr(1, "dracut", "-f", "--kver", kernelVersion, "-a", dracutModuleName)
+		return err
+	})
+	if err != nil {
+		return fmt.Errorf("failed to build dracut module - (%s):\n%w", dracutModuleName, err)
+	}
+
+	return nil
+}
+
+func updateMarinerCfgWithInitramfs(imageChroot *safechroot.Chroot) error {
+	var err error
+
+	initramfsPath := filepath.Join(imageChroot.RootDir(), "boot/initramfs-*")
```

Comment:
Shouldn't hard-code directory separator "/" in combination with filePath.Join usage.

Does this work?

filepath.Join(imageChroot.RootDir(), "boot", "initramfs-*")

Reply:
Also below